internal/daemon/api_files.go

```diff
@@ -250,7 +250,7 @@ func fileInfoToResult(fullPath string, info os.FileInfo, userCache, groupCache m
 		Name:         pathpkg.Base(fullPath),
 		Type:         fileModeToType(mode),
 		Size:         psize,
-		Permissions:  fmt.Sprintf("%03o", mode.Perm()),
+		Permissions:  fmt.Sprintf("%04o", mode.Perm()),
```

Comment:
Similar to above, I'd prefer this to be `%03o` so it's 3 digit by default and older, tightly-validating clients don't reject it. I realize this probably won't happen in reality, because the main client (Python Operator Framework) looks like it handles any number of digits, but better to be safe and backwards-compatible. Also, go's `mode.Perm()` will never return more than 3 digits now anyway, because `ModePerm FileMode = 0777`.